tests/ui/single_char_pattern.fixed

```diff
@@ -18,9 +18,9 @@ fn main() {
     //
     // We may not want to suggest changing these anyway
     // See: https://github.com/rust-lang/rust-clippy/issues/650#issuecomment-184328984
```

Comment:
Oh I did not see this comment when testing the change.

So, it seems this does not ICE anymore, which is good :) After taking a look at the mentioned issue, I think we can just remove this comment. There was no benchmark to back the affirmation that the methods taking a single character are slower with multi-byte characters, but there were benchmarks showing the opposite for the single character version in general. Moreover, it's been 4 years, so the claim may very well be outdated. 

I would favour consistency here, what do you think?